### PR TITLE
Reload modules on pip install

### LINKED ISSUE #611: Restore reloading of modules on pip install

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
Fixes #611

### CODE REVIEW

File: src/LanguageServer/Impl/LanguageServer.cs [hunk 1]
```diff
@@ -377,19 +380,29 @@ public Task ExtensionCommand(JToken token, CancellationToken cancellationToken)
...
                 var interpreter = _services.GetService<IPythonInterpreter>();
+                var logger = _services.GetService<ILogger>();
                 _pathsWatcher = new PathsWatcher(
                     _initParams.initializationOptions.searchPaths,
-                    () => interpreter.ModuleResolution.ReloadAsync(cancellationToken).DoNotWait(),
-                    _services.GetService<ILogger>()
-                 );
+                    () => {
+                        logger.Log(TraceEventType.Information, Resources.ReloadingModules);
+                        interpreter.ModuleResolution.ReloadAsync(cancellationToken).ContinueWith(t => {
```
Comment (AlexanderSher) [inline src/LanguageServer/Impl/LanguageServer.cs]: Should we clean the `ModuleResolutionBase._modules` when we do `ReloadAsync`?

File: src/Analysis/Ast/Impl/Modules/PythonModule.cs [hunk 2]
```diff
@@ -155,7 +155,6 @@ internal PythonModule(ModuleCreationOptions creationOptions, IServiceContainer s
                 return collection.Contents
                     .OfType<IPythonConstant>()
                     .Select(c => c.GetString())
-                    .ExcludeDefault()
```
Comment (AlexanderSher) [inline src/Analysis/Ast/Impl/Modules/PythonModule.cs]: Why?

Comment (MikhailArkhipov) [reply src/Analysis/Ast/Impl/Modules/PythonModule.cs]: Next condition is !string.isNullOfEmpty

File: src/LanguageServer/Impl/LanguageServer.cs (hunk 1 above)
Comment (MikhailArkhipov) [reply src/LanguageServer/Impl/LanguageServer.cs]: `ReloadAsync` technically should be doing it, let me check

**Pull request merged**